packages/merge-on-green/src/merge-logic.ts

```diff
@@ -141,166 +172,165 @@ mergeOnGreen.hasMOGLabel = async function hasMOGLabel(
   }
 };
 
-mergeOnGreen.requiredChecksByLanguage = async function requiredChecksByLanguage(
+/**
+ * Function grabs the required checks of the master branch
+ * @param owner of pr (from Watch PR)
+ * @param repo of pr (from Watch PR)
+ * @param github unique installation id for each function
+ * @returns a string array of the names of the required checks
+ */
+mergeOnGreen.getBranchProtection = async function getBranchProtection(
+  owner: string,
+  repo: string,
   github: GitHubAPI
-): Promise<RequiredChecksByLanguage | null> {
+): Promise<string[]> {
   try {
-    const configFile = (
-      await github.repos.getContents({
-        owner: 'googleapis',
-        repo: 'sloth',
-        path: 'required-checks.json',
+    const branchProtection = (
+      await github.repos.getBranchProtection({
+        owner,
+        repo,
+        branch: 'master',
       })
-    ).data as { content?: string };
-    return JSON.parse(
-      Buffer.from(configFile.content as string, 'base64').toString('utf8')
-    ) as RequiredChecksByLanguage;
+    ).data.required_status_checks.contexts;
+    console.log(
+      `checking branch protection for ${owner}/${repo}: ${branchProtection}`
+    );
+    return branchProtection;
   } catch (err) {
-    return null;
+    return [];
   }
 };
 
-/*
-The function above gets the required checks for each repo based on
-language. However, since we only have the repo name from the payload,
-we need to figure out what language the repo name attaches to. Grabbing
-this file tells us that info.
-*/
-mergeOnGreen.mapReposToLanguage = async function mapReposToLanguage(
-  github: GitHubAPI
-): Promise<Language[]> {
+/**
+ * Function grabs the statuses that have run for a given Sha
+ * @param owner of pr (from Watch PR)
+ * @param repo of pr (from Watch PR)
+ * @param github unique installation id for each function
+ * @param headSha the head sha commit
+ * @returns an array of Check Statuses that has their names and statuses
+ */
+mergeOnGreen.getStatusi = async function getStatusi(
+  owner: string,
+  repo: string,
+  github: GitHubAPI,
+  headSha: string,
+  num: number
+): Promise<CheckStatus[]> {
+  const start = Date.now();
   try {
-    const configFile = (
-      await github.repos.getContents({
-        owner: 'googleapis',
-        repo: 'sloth',
-        path: 'repos.json',
+    const data = (
```

Comment:
You can shorten this to `const {data} = await github.repos.listStatusesForRef....`